### PR TITLE
Clean up `grakn-graph` property tests and generators

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

File: grakn-test-tools/src/main/java/ai/grakn/generator/AbstractOntologyConceptGenerator.java [hunk 1]
```diff
@@ -112,26 +100,21 @@ private final boolean includeNonMeta() {
...
 
-    final AbstractOntologyConceptGenerator<T> excludeAbstract() {
-        includeAbstract = Optional.of(false);
-        return this;
-    }
-
     public final void configure(Meta meta) {
-        this.meta = Optional.of(meta.value());
+        Preconditions.checkArgument(
+                !this.meta.isPresent() || this.meta.get(), "Cannot specify parameter is both meta and non-meta");
```
Comment (fppt) [inline grakn-test-tools/src/main/java/ai/grakn/generator/AbstractOntologyConceptGenerator.java]: > Cannot specify parameter is both meta and non-meta

What?? How can this happen?

File: grakn-test-tools/src/main/java/ai/grakn/generator/AbstractTypeGenerator.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    AbstractTypeGenerator(Class<T> type) {
+        super(type);
+    }
+
+    private boolean includeAbstract(){
+        return includeAbstract.orElse(true);
+    }
+
+    final AbstractOntologyConceptGenerator<T> excludeAbstract() {
```
Comment (fppt) [inline grakn-test-tools/src/main/java/ai/grakn/generator/AbstractTypeGenerator.java]: `includeAbstract()` returns a boolean while this one returns the generator itself. Sad :(

File: grakn-test-tools/src/main/java/ai/grakn/generator/AbstractOntologyConceptGenerator.java (hunk 1 above)
Comment (aelred) [inline grakn-test-tools/src/main/java/ai/grakn/generator/AbstractOntologyConceptGenerator.java]: Haha, just if you write a property test with parameter: `@Meta @NonMeta Concept concept` :P

File: grakn-test-tools/src/main/java/ai/grakn/generator/AbstractTypeGenerator.java (hunk 2 above)
Comment (aelred) [inline grakn-test-tools/src/main/java/ai/grakn/generator/AbstractTypeGenerator.java]: I should name them better... one is a getter, the other a setter.

**Pull request merged**